pkg/devfile/devfile.go

```diff
@@ -0,0 +1,25 @@
+package devfile
+
+// DevfileHeader describes the structure of the devfile-specific top-level fields
+// that are not part of the K8S API structures
+type DevfileHeader struct {
+	// Devfile schema version
+	// +kubebuilder:validation:Pattern=^([2-9])\.([0-9]+)\.([0-9]+)(\-[0-9a-z-]+(\.[0-9a-z-]+)*)?(\+[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?$
+	SchemaVersion string `json:"schemaVersion"`
```

Comment:
> will make it easier to simply unmarshal any devfile 2+ go get header fields from raw Json or Yaml.

Does it make sense to declare unmarshaling somewhere here? Or devfile/parser is a better place for such stuff and one day it will appear there?